### PR TITLE
Add idempotence tests

### PR DESCRIPTION
**Pull request by sbarzowski**
Just run formatter again on golden and see if something changes.
It shouldn't.

### CODE REVIEW

**Comment by sbarzowski** [comment]
This commit is a side-effect of my fights with weird paragraph fodder problems. Cool part is that it uses existing test data, so we have more comprehensive test-suite with little additional effort once we merge this. It's important to keep formatter idempotent, yet even very serious ones (like gofmt) have occasional problems with that. So IMO it's worth it to check.

@lwander @sparkprime

**Comment by sparkprime** [comment]
:+1:

File: test_suite/run_fmt_idempotence_tests.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    GOLDEN_OUTPUT="${ORIG_OUTPUT}"
+    GOLDEN_KIND="PLAIN"
+
+    if [ -r "$TEST.fmt.golden" ] ; then
+        GOLDEN_OUTPUT=$(cat "$TEST.fmt.golden")
+        FMT_FILE="$TEST.fmt.golden"
+    fi
+
+    if [ $(echo "$TEST" | cut -b 1-12) == "error.parse." ] ; then
+        continue  # No point testing these
```
Comment (lwander) [inline test_suite/run_fmt_idempotence_tests.sh]: Maybe it's worth testing that no regression in parsing/import formatting gets introduced that causes these to misbehave?

Comment (sbarzowski) [inline test_suite/run_fmt_idempotence_tests.sh]: We don't get to format `error.parse.*` at all (formatting happens after parsing), so there is no interaction with the formatter. And we check that the error is what we expect in run_tests.sh. So I think we're covered here.

Unless what you meant is running golden files and verifying that the output matches the output of original files. That could help with things like variable shadowing in imports:
```
local b = import "b.jsonnet";
local a = import "a.jsonnet";
local b = import "a.jsonnet";
```
But I guess that would be another script.

**Pull request merged**